packages/client/bin/cli.ts

```diff
@@ -222,12 +234,56 @@ async function run() {
       chain: genesisParams.name,
       customChains: [[genesisParams, genesisState]],
     })
-  }
-  // Use default common configuration for specified `chain` if no custom parameters specified
-  else {
+  } else {
+    // Use default common configuration for specified `chain` if no custom parameters specified
     common = new Common({ chain: chain, hardfork: Hardfork.Chainstart })
   }
 
+  if (args.mine) {
+    if (!args.customChain && !args.gethGenesis) {
+      throw new Error('Currently mining is only supported for custom chains')
```

Comment:
the reason for this is that we wouldn't have a valid signer so we can't actually seal a block on an existing PoA chain, it would have to be a custom genesis block with a signer that is configured to the one passed into `--unlock`

Reply:
Ah, that can't be the reason for this restriction, the decision if someone has a signer on a chain is happening on some other level. Let's say Erigon has no signer on Rinkeby, that wouldn't mean that they would restrict their client in a similar way and say "we don't support Rinkeby".

At the end the list of signers in Clique is dynamic and it can theoretically be that one of our signers will be voted into the list of active signers for Clique in the future. If we are brave, we can also join the next ephemeral testnet as a signer. 😄

So this is very much a limitation happing at the wrong place and putting unnecessary limitations to the client.

Reply:
(or am I missing the point here?)

Reply:
> Let's say Erigon has no signer on Rinkeby, that wouldn't mean that they would restrict their client in a similar way and say "we don't support Rinkeby".

Makes sense, so in `assembleBlock()` instead I could check the unlocked signer account against the `activeSignersList` and return with an error if not present, since otherwise an error will be thrown when putting the block in our own chain since it won't match signer validation.

Reply:
Yes, I guess that makes more sense. 🙂 👍